sycl/source/detail/platform_impl.cpp

```diff
@@ -211,7 +211,7 @@ static std::vector<DevDescT> getAllowListDesc() {
   return DecDescs;
 }
 
-enum MatchState { UNKNOWN, MATCH, NOMATCH };
+enum MatchState { UNKNOWN, NOMATCH };
```

Comment:
It's hard to understand if the device matches (i.e. passes the filter) when one can choose either "not matched" or "unknown" matching state.
Looks like it needs some renaming: unknown -> match.